src/runtime/opencl/opencl_common.h

```diff
@@ -216,7 +216,8 @@ class OpenCLWorkspace : public DeviceAPI {
   // Initialzie the device.
   void Init(const std::string& type_key, const std::string& device_type,
             const std::string& platform_name = "");
-  virtual void Init() { Init("opencl", "gpu"); }
+  // Find accelerator first, TODO: Make a proper way to chose between device types
+  virtual void Init() { Init("opencl", "accelerator"); }
```

Comment:
I think we could have one discussion here. That is whether we should let `accelerator` be the first priority for OpenCL or we should do this like Intel FPGA (AOCL), for accelerator we create one new device type.

Reply:
I agree there needs to have a discussion. I think that putting accelerator in the default OpenCL is a necessity because it will allow people who use custom or currently unsupported accelerators to use TVM easily. However because it will use CUDA schedules, the performances may suffer so maybe putting accelerators with a low priority would be better.

At the end, the present system is in my opinion flawed because it does not let the user chose which type of device to prioritize for OpenCL. Maybe we should split the current OpenCL target to three (or four with custom?) targets like "opencl-cpu", "opencl-gpu" and "opencl-accelerator". Or we could use a target attribute to do it. What do you think about it?

Reply:
When we lift the `accelerator` as the highest level, one big problem is we will use CUDA schedule as you have mentioned. I suspect we will get wrong result sometimes which is worse than poor performance (I need your expert experience to help me make sure this).

Current issue is we don't have one elegant way to let us accept device type when we `Init`.  If we want to support it, better way is we could expose it to let users choose. However, I prefer current Intel FPGA / Xilinx way. Intel FPGA creates `AOCL` like this:
```cpp
void AOCLWorkspace::Init() {
  OpenCLWorkspace::Init("aocl", "accelerator", "Intel(R) FPGA SDK for OpenCL(TM)");
}
```
Xilinx is:
```cpp
void SDAccelWorkspace::Init() { OpenCLWorkspace::Init("sdaccel", "accelerator", "Xilinx"); }
``` 
All different things you could do in your own device api. Like `class YourOwnTarget : public OpenCLWorkspace`.